generators/csharp/codegen/src/ast/MethodInvocation.ts

```diff
@@ -49,11 +61,16 @@ export class MethodInvocation extends AstNode {
         this.configureAwait = configureAwait ?? false;
         this.generics = generics ?? [];
         this.multiline = multiline ?? false;
+        this.isAsyncEnumerable = isAsyncEnumerable ?? false;
     }
 
     public write(writer: Writer): void {
-        if (this.async) {
-            writer.write("await ");
+        if (this.isAsyncEnumerable) {
```

Comment:
I think `await foreach(var item in` is a stretch for it being considered method-invocation. Let's create a different AST node for iteration or do something else, but not put it inside this.

Reply:
Sure, we'll followup later on this.